bundles/specmate-testspecification/src/com/specmate/testspecification/internal/generators/ProcessTestCaseGenerator.java

```diff
@@ -104,66 +139,211 @@ private TestParameter createAndAddTestParameterIfNecessary(String testParameterN
 		}
 		return testParameter;
 	}
+	
+	/**
+	 * @param condition the condition to search for
+	 * @return all conditions that should not be taken if the actual condition is selected
+	 */
+	private Set<String> getConflictingConditions(String condition) {
+		Set<IModelNode> decisionNodes = nodes.stream().filter(n -> hasCondition(n, condition)).collect(Collectors.toSet());
+		Set<String> conflictingConditions = decisionNodes.stream()
+				.flatMap(n -> n.getOutgoingConnections().stream())
+				.filter(c -> !hasCondition(c, condition))
+				.map(c -> (ProcessConnection) c)
+				.map(c -> c.getCondition())
+				.collect(Collectors.toSet());
+		return conflictingConditions;
+	}
+	
+	private static boolean hasCondition(IModelNode node, String condition) {

```

Comment:
There is also a non-static hasCondition method, I find this a bit confusing to have static and non-static methods with the same name in different places in the file